### PR TITLE
[WIP] Amber mod

### PR DESCRIPTION
**Pull request by theraven262**
- [x] Import amber nodes, craftitems, crafts and oregen
- [x] Adjust the mapgen and crafting values (game balance)

### CODE REVIEW

**Comment by theraven262** [comment]
Further balancing oregen and crafting can be done after the merge.

**Comment by neinwhal** [comment]
Looks good, squashing and merging.

**Comment by wrynvera** [comment]
I'm ok with the amber mod, but I would like to see more approvals to this before merging it as a core ore and node.

File: mods/blocks/glass.lua [hunk 1]
```diff
@@ -26,10 +26,53 @@ minetest.register_node("blocks:obsidian_glass", {
 	groups = {cracky = 3},
 })
 
+-- Amber Glass
+
+minetest.register_node("blocks:amber_glass", {
+	description = S("Amber Glass"),
+	drawtype = "allfaces",
+	tiles = {"blocks_amber_glass.png"},
+	use_texture_alpha = true,
```
Comment (wrynvera) [inline mods/blocks/glass.lua]: deprecated

File: mods/blocks/glass.lua [hunk 2]
```diff
@@ -26,10 +26,53 @@ minetest.register_node("blocks:obsidian_glass", {
...
+	is_ground_content = false,
+	groups = {cracky = 3, oddly_breakable_by_hand = 3},
+	sounds = default.node_sound_glass_defaults(),
+})
+
+minetest.register_node("blocks:amber_medieval_glass", {
+	description = S("Medieval Amber Glass"),
+	drawtype = "allfaces",
+	tiles = {"blocks_amber_glass_medieval.png"},
+	use_texture_alpha = true,
```
Comment (wrynvera) [inline mods/blocks/glass.lua]: deprecated

File: mods/mapgen/oregen.lua [hunk 3]
```diff
@@ -576,17 +597,6 @@ function mapgen.register_ores()
...
-		ore_type       = "scatter",
-		ore            = "blocks:stone_with_coal",
-		wherein        = "blocks:stone",
-		clust_scarcity = 8 * 8 * 8,
-		clust_num_ores = 8,
-		clust_size     = 3,
-		y_max          = 64,
-		y_min          = -127,
-	})
-
```
Comment (wrynvera) [inline mods/mapgen/oregen.lua]: any reason why this was removed?

Comment (theraven262) [reply mods/mapgen/oregen.lua]: Accidental, this PR shouldn't remove any default code.

**Review by lumberjackgames — COMMENTED** [review]
Overall I think adding this mod is fine, but I think we should look closer at the ore mapgen in proportion to other ores. If I am not mistaken, as is this will be an extremely common byproduct of mining (much more than coal) with both generous scatter and sheet ores. If that is what is wanted, fine, but it is an issue that merits some thought I think.

**Comment by theraven262** [comment]
> Overall I think adding this mod is fine, but I think we should look closer at the ore mapgen in proportion to other ores. If I am not mistaken, as is this will be an extremely common byproduct of mining (much more than coal) with both generous scatter and sheet ores. If that is what is wanted, fine, but it is an issue that merits some thought I think.

The mapgen balance will be done after we figure out what crafting recipes are going to be. And yeah I made it really common here, but thats a placeholder.

**Comment by wrynvera** [comment]
I think raven's response is fair. Also considering we haven't even decided on a mapgen yet.
I'm okay with merging once all the changes i requested in my review is made.

**Comment by lumberjackgames** [comment]
Once the conflicts are worked out I'm good to merge

**Pull request merged**